include/deal.II/fe/fe_system.h

```diff
@@ -422,6 +423,21 @@ class FESystem : public FiniteElement<dim,spacedim>
             const std::vector<unsigned int>                   &multiplicities);
 
   /**
+   * Constructor taking an arbitrary number of parameters of type
+   * <code>std::pair<std::unique_ptr<FiniteElement<dim, spacedim>>, unsigned int></code>.
+   * In combination with FiniteElement::operator^, this allows to construct FESystem objects
+   * as follows:
+   * @code
+   *   FESystem<dim, spacedim> = {FiniteElementType1^{n1}, FiniteElementType2^{n2},...};
+   * @endcode
+   */
+  template <class... FESystems,
+            typename = typename std::enable_if<all_same_as<typename std::decay<FESystems>::type...,
+                                                           std::pair<std::unique_ptr<FiniteElement<dim, spacedim>>,
+                                                               unsigned int>>::value>::type>
+  FESystem (FESystems&& ... fe_systems);
```

Comment:
Could you use `std::initializer_list<std::pair<...,...>>` here instead? I think if you did it this way, you'd force the compiler to do the conversion right away, right?

Reply:
Alternatively, one could put the `enable_if` part into a `static_assert` inside the implementation of the constructor. This allows giving reasonable error messages, rather than the compiler just saying that there is no matching constructor. The downside is that if you do it this way, you can have only one variadic constructor.

Reply:
With the current approach we can write
```
FESystem<dim> fe_system(FE_Q<dim>(2)^dim, FE_Q<dim>(1)^1);
```
as well as
```
FESystem<dim> fe_system {FE_Q<dim>(2)^dim, FE_Q<dim>(1)^1};
```
while with a `std::initializer_list` we can only do the latter. Also you can't do perfect forwarding with `std::initializer_list` (although we probably don't really benefit from this at the moment).
Do you feel strong about this?

Putting the `enable_if` part into a `static_assert` would lead to problems due to the universal reference eating everything. Thus, we would have to remove all the other constructors which we probably don't want to.

Reply:
I think you could also write
```
  FESystem<dim> fe_system({FE_Q<dim>(2)^dim, FE_Q<dim>(1)^1});
```
My worry is that with the current system you don't get particularly good error messages if you get anything wrong -- say, if you forget the `^1`. With the `std::enable_if`, the compiler is just going to say that no such constructor is available, but it will not be able to say something like 
```
  fe_system:325: candidate '...' can not be used because argument 5 of type ABC cannot be
     converted to type DEF
```
That's the downside of using `std::enable_if` -- you just don't get good error messages. You only get those if you have known types.